From e859580ada70b96107d9530f4c80d23863867a9d Mon Sep 17 00:00:00 2001
From: Tony Mak <tonymak@google.com>
Date: Thu, 14 Dec 2017 12:40:07 +0000
Subject: [PATCH 038/306] clearCallingIdentity before calling into
 getPackageUidAsUser

Fix: 70585244

Test: Enable any accessibility service -> inflate work profile
      -> Tap on any work app -> no longer crash

Test: cts-tradefed run cts-dev --module DevicePolicyManager --test com.android.cts.devicepolicy.CrossProfileAppsHostSideTest.testPrimaryUserToManagedProfile
Change-Id: I80d18f4e2ab76a228cb0aa2c8312c323a9b5c84d
(cherry picked from commit 1232d583bb54fac88b510b72d8dbc79ced36f0f4)
---
 .../server/accessibility/AccessibilityManagerService.java      | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/services/accessibility/java/com/android/server/accessibility/AccessibilityManagerService.java b/services/accessibility/java/com/android/server/accessibility/AccessibilityManagerService.java
index ac0cdd7cf99..f417f0b4bf5 100644
--- a/services/accessibility/java/com/android/server/accessibility/AccessibilityManagerService.java
+++ b/services/accessibility/java/com/android/server/accessibility/AccessibilityManagerService.java
@@ -2962,11 +2962,14 @@ public class AccessibilityManagerService extends IAccessibilityManager.Stub
         }
 
         private boolean isValidPackageForUid(String packageName, int uid) {
+            final long token = Binder.clearCallingIdentity();
             try {
                 return uid == mPackageManager.getPackageUidAsUser(
                         packageName, UserHandle.getUserId(uid));
             } catch (PackageManager.NameNotFoundException e) {
                 return false;
+            } finally {
+                Binder.restoreCallingIdentity(token);
             }
         }
 
-- 
2.17.1

